.github/workflows/gh-pages.yml

```
@@ -0,0 +1,29 @@
+name: github pages
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: "0.4.21"
+
+      - name: Build book
+        run: mdbook build specs
+
+      - name: Deploy main
+        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
```

[nit] I'm pretty sure you just need to check the `github.event_name` since we are already filtering to only triggering on the `main` branch for `push` events up top. But if you'd prefer to keep this here as an additional level of protection that is fine. 

```suggestion
        if: github.event_name == 'push'
```

---

good point! Will adopt your suggestion